motortorquecalculation/alt_gain.py

```diff
@@ -81,96 +126,181 @@ def speed_req(self, angle, torque):
-            efficiency = 0 
-        total_energy += pre_eff_energy + wattage * (dist[point - 1] - dist[point - 2]) / data[point][0]
-        if total_energy > 5.04 * 10 ** 7:
-            breakpoint.append(dist[point - 1])
-            full_length = False
-            break
-    total_energies.append(total_energy)
-    if full_length:
-        breakpoint.append(dist[-1])
-break_in_km = [x / 1000 for x in breakpoint]
-normalized_wattages = [x - baseline for x in wattages]
-plt.plot(normalized_wattages, break_in_km)
-plt.xlabel('Added Energy Use (W)')
-plt.ylabel('Range (km)')
-plt.show()
+
+def generate_parameters(alt, lat, lon):
+    """
+    Given the raw data, generate parameters (distances, delta altitude, angles and climb.
+    @param alt: Altitude data (
+    @type alt: List[float]
+    @param lat: Latitude
+    @type lat: List[float]
+    @param lon: Longitudes
+    @type lon: List[float]
+    @return: Distances, delta in altitudes, angles and climb.
+    @rtype:(List[float], List[float], List[float], List[float])
+    """
+    dist = [0]
+    # TODO: Unused variable: dalt.
+    dalt = [0]
+    climb = [0]
+    angles = [0]
+    for i in range(len(alt) - 1):
+        dist.append(
+            haversine((lat[i], lon[i]), (lat[i + 1], lon[i + 1]), unit=Unit.METERS)
+            + dist[i]
+        )
+        dalt = alt[i + 1] - alt[i]
+        angle = dalt / (dist[i + 1] - dist[i])
+        angles.append(angle)
+        climb.append((180 / pi) * arctan(angle))
+    return dist, dalt, climb, angles
+
+
+def main():
+    """
+    Plots a graph detailing range (km) over added energy use (W).
+    """
+    parser = argparse.ArgumentParser(
+        description="Choose a map",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--map", help="Required: Map to pick to race on: [WSC|ASC]", required=True
+    )
+    parser.add_argument(
+        "--solar", help="Energy we receive from the solar panel", default=0, type=float
+    )
+    parser.add_argument(
+        "--weight", help="Weight of the car in kg", default=720, type=float
+    )
+    args = parser.parse_args()
+
+    if args.map == "WSC":
+        csv_file = pd.read_csv("wsc_elevation.csv")
+        lon = csv_file.Longitude.to_list()
+        lat = csv_file.Latitude.to_list()
+        alt = csv_file["Elevation (m)"].to_list()
+        speed_req = kph_to_ms(80)
+    elif args.map == "ASC":
+        csv_file = pd.read_csv("ASC2018.csv")
+        lon = csv_file.lon.to_list()
+        lat = csv_file.lat.to_list()
+        alt = csv_file.alt.to_list()
+        speed_req = kph_to_ms(60)
+
+    car = Car(args.weight, 0.15, 0.0015)
+
+    dist, _, climb, angles = generate_parameters(alt, lat, lon)
+    avg = mean(x for x in climb if x > -(180 / pi) * 0.0085035)
+    print("Car torque required per wheel (Nm): ", car.torque_req(avg, speed_req))
+
+    # TODO: Another unused variable: torques. It is overwritten to [] in the loop below right away.
```

Comment:
I'd like to deal with this TODO before we merge

Reply:
There's a couple of other unused variables as well. I kept them all in since I thought it might be the case that you might be changing the script and generating different graphs/etc.